Stimuli/Tones.py

```diff
@@ -16,17 +16,23 @@ class Tones(RPScreen, dj.Manual):
     cond_tables = ['Tones']
     required_fields = ['tone_duration', 'tone_frequency']
     default_key = {'tone_volume': 50, 'tone_pulse_freq': 0}
-
-    def stop(self):
-        super(RPScreen, self).stop()
-        self.exp.interface.stop_sound()
-        
+       
     def start(self):
-        tone_duration = self.curr_cond['tone_duration']
         tone_frequency = self.curr_cond['tone_frequency']
         tone_volume = self.curr_cond['tone_volume']
         tone_pulse_freq=self.curr_cond['tone_pulse_freq']
         if 0< self.curr_cond['tone_pulse_freq']<10 :
             raise ValueError('Tone pulse frequency cannot be between zero and 10Hz (not including)')
-        self.exp.interface.give_sound(tone_frequency, tone_duration, tone_volume, tone_pulse_freq)
-        super().start()
\ No newline at end of file
+        self.exp.interface.give_sound(tone_frequency, tone_volume, tone_pulse_freq)
+        super().start()
+
+    def present(self):
+        if self.timer.elapsed_time() > self.curr_cond['tone_duration'] and self.isrunning:
+            self.isrunning = False
+            self.stop()
+            super().ready_stim()
```

Comment:
why do you want the ready_stim here?

Reply:
removed, ready_stim is defined in the experiment